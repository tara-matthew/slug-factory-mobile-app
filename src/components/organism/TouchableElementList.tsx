import React from "react";
import { View } from "react-native";
import TouchableLink from "../molecule/TouchableLink";
import { ITouchableElementListProps, ITouchableLinkProps } from "../../contracts/TouchableLink";

const TouchableElementList = ({ items }: ITouchableElementListProps) => {
    return (
        <View>
            {items.map((item: ITouchableLinkProps, index: number) => (
                <TouchableLink to={ item.to } title={ item.title } key={ index } />
            ))}
        </View>
    );
};

export default TouchableElementList;
