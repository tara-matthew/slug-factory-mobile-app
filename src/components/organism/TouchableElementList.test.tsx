import React from "react";
import { render } from "@testing-library/react-native";
import TouchableElementList from "./TouchableElementList";

it("renders correctly", () => {
    const profileMock = jest.fn();

    const items = [
        {
            to: profileMock,
            title: "My profile",
        },
        {
            to: profileMock,
            title: "Edit profile",
        },
        {
            to: profileMock,
            title: "My prints",
        },
    ];
    const { toJSON } = render(<TouchableElementList items={ items } />);
    expect(toJSON()).toMatchSnapshot();
});

it.todo("displays the correct text");
