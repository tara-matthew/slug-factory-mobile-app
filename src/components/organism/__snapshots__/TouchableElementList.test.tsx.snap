// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View>
      <View
        className="flex flex-row justify-between items-center w-full"
      >
        <Text
          className="py-5 px-10 text-lg"
        >
          My profile
        </Text>
        Icon: chevron-right, Size: 20, Color: undefined
      </View>
      <View
        className="w-full"
        style={
          {
            "backgroundColor": "gray",
            "height": 1,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View>
      <View
        className="flex flex-row justify-between items-center w-full"
      >
        <Text
          className="py-5 px-10 text-lg"
        >
          Edit profile
        </Text>
        Icon: chevron-right, Size: 20, Color: undefined
      </View>
      <View
        className="w-full"
        style={
          {
            "backgroundColor": "gray",
            "height": 1,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View>
      <View
        className="flex flex-row justify-between items-center w-full"
      >
        <Text
          className="py-5 px-10 text-lg"
        >
          My prints
        </Text>
        Icon: chevron-right, Size: 20, Color: undefined
      </View>
      <View
        className="w-full"
        style={
          {
            "backgroundColor": "gray",
            "height": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
